master/buildbot/test/unit/test_scripts_cleanupdb.py

```diff
@@ -86,6 +99,12 @@ def test_cleanup_bad_config(self):
 
     @defer.inlineCallbacks
     def test_cleanup_bad_config2(self):
+        # test may use mysql or pg if configured in env
+        if "BUILDBOT_TEST_DB_URL" not in os.environ:
+
+            patch_environ(self, "BUILDBOT_TEST_DB_URL", "sqlite:///" + os.path.join(self.origcwd,
+                                                                             "basedir", "state.sqlite"))
+
```

Comment:
I'm not sure if there is really any value in factoring these two lines out.

I think a better solution would be to actually stop mutating the environment, and instead just pass some parameters around. But `test_cleanup` calls `setUpRealDatabase` which also inspects the environment, and the tests appear to depend on this specific value, rather than the one `setUpRealDatabase`.

I'm not directly interested in cleaning up these tests completely, but I think that this change is an improvement on the current state.